### PR TITLE
Add mention of geo_shape support in geotile and geohash grid agg docs

### PR DESCRIPTION
**Pull request by talevy**
Previously, geo_shape support was only mentioned in a dedicated x-pack
section. This may be misleading, as the introductory paragraph only
mentions geo_point.

### CODE REVIEW

File: docs/reference/aggregations/bucket/geotilegrid-aggregation.asciidoc [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [[search-aggregations-bucket-geotilegrid-aggregation]]
 === GeoTile Grid Aggregation
 
-A multi-bucket aggregation that works on `geo_point` fields and groups points into
+A multi-bucket aggregation that works on `geo_point` and `geo_shape` fields and groups points into
```
Comment (nreese) [inline docs/reference/aggregations/bucket/geotilegrid-aggregation.asciidoc:4]: "and groups points into buckets" leaves out how geo_shapes work. How about enhancing with text how geo_shapes are matched into buckets as well. Maybe something like "and groups points, lines, and polygons into buckets"?

File: docs/reference/aggregations/bucket/geohashgrid-aggregation.asciidoc [hunk 2]
```diff
@@ -4,7 +4,7 @@
 <titleabbrev>Geohash grid</titleabbrev>
 ++++
 
-A multi-bucket aggregation that works on `geo_point` fields and groups points into buckets that represent cells in a grid.
+A multi-bucket aggregation that works on `geo_point` and `geo_shape` fields and groups points, lines and shapes into buckets that represent cells in a grid.
```
Comment (jrodewig) [inline docs/reference/aggregations/bucket/geohashgrid-aggregation.asciidoc:7]: I tried to streamline this a little bit, but feel free to ignore if wanted.
```suggestion
A multi-bucket aggregation that groups <<geo-point,`geo_point`>> and
<<geo-shape,`geo_shape`>> values into buckets that represent a grid.
```

File: docs/reference/aggregations/bucket/geotilegrid-aggregation.asciidoc [hunk 3]
```diff
@@ -4,8 +4,9 @@
 <titleabbrev>Geotile grid</titleabbrev>
 ++++
 
-A multi-bucket aggregation that works on `geo_point` fields and groups points into
-buckets that represent cells in a grid. The resulting grid can be sparse and only
+A multi-bucket aggregation that works on `geo_point` and `geo_shape` fields and groups
+points, lines and shapes into buckets that represent cells in a grid.
+The resulting grid can be sparse and only
```
Comment (jrodewig) [inline docs/reference/aggregations/bucket/geotilegrid-aggregation.asciidoc:9]: Same as above.

File: docs/reference/aggregations/bucket/geotilegrid-aggregation.asciidoc [hunk 4]
```diff
@@ -26,9 +27,9 @@ costly in terms of RAM and result sizes. Please see the example below on how
 to first filter the aggregation to a smaller geographic area before requesting
 high-levels of detail.
 
-The specified field must be of type `geo_point` (which can only be set
-explicitly in the mappings) and it can also hold an array of `geo_point`
-fields, in which case all points will be taken into account during aggregation.
+The specified field must be of type `geo_point` or `geo_shape` (which can only be
+set explicitly in the mappings). It can also hold an array of `geo_point` fields,
+in which case all points will be taken into account during aggregation.
```
Comment (jrodewig) [inline docs/reference/aggregations/bucket/geotilegrid-aggregation.asciidoc:32]: Same as above.

File: docs/reference/aggregations/bucket/geohashgrid-aggregation.asciidoc [hunk 5]
```diff
@@ -15,7 +15,9 @@ Geohashes used in this aggregation can have a choice of precision between 1 and
 WARNING: The highest-precision geohash of length 12 produces cells that cover less than a square metre of land and so high-precision requests can be very costly in terms of RAM and result sizes.
 Please see the example below on how to first filter the aggregation to a smaller geographic area before requesting high-levels of detail.
 
-The specified field must be of type `geo_point` (which can only be set explicitly in the mappings) and it can also hold an array of `geo_point` fields, in which case all points will be taken into account during aggregation.
+The specified field must be of type `geo_point` or `geo_shape` (which can only be
+set explicitly in the mappings). It can also hold an array of `geo_point` fields,
+in which case all points will be taken into account during aggregation.
```
Comment (jrodewig) [inline docs/reference/aggregations/bucket/geohashgrid-aggregation.asciidoc:20]: I wonder if this paragraph is needed. We already state that the agg works on `geo_point` and `geo_shape` fields. The note about explicit mappings seems more relevant to the field types than the agg.

Do users commonly use `geo_point` to store arrays? If so, would they assume array values are supported (as they are elsewhere in ES)?

I included a suggestion if we decide to keep the para. Feel free to ignore if wanted.
```suggestion
You can only use `geotile_grid` to aggregate an explicitly mapped `geo_point` or
`geo_shape` field. If the `geo_point` field contains an array, `geotile_grid`
aggregates all the array values.
```

**Review by jrodewig — APPROVED** [review]
LGTM. I left some optional suggestions, but feel free to ignore those if wanted. Thanks for fixing this!

**Comment by talevy** [comment]
thanks for the suggestions @jrodewig. included!

File: docs/reference/aggregations/bucket/geohashgrid-aggregation.asciidoc [hunk 6]
```diff
@@ -15,7 +16,9 @@ Geohashes used in this aggregation can have a choice of precision between 1 and
 WARNING: The highest-precision geohash of length 12 produces cells that cover less than a square metre of land and so high-precision requests can be very costly in terms of RAM and result sizes.
 Please see the example below on how to first filter the aggregation to a smaller geographic area before requesting high-levels of detail.
 
-The specified field must be of type `geo_point` (which can only be set explicitly in the mappings) and it can also hold an array of `geo_point` fields, in which case all points will be taken into account during aggregation.
+You can only use `geotile_grid` to aggregate an explicitly mapped `geo_point` or
+`geo_shape` field. If the `geo_point` field contains an array, `geotile_grid`
+aggregates all the array values.
```
Comment (jrodewig) [inline docs/reference/aggregations/bucket/geohashgrid-aggregation.asciidoc:21]: Fixes a copy/paste error on the agg name
```suggestion
You can only use `geohash_grid` to aggregate an explicitly mapped `geo_point` or
`geo_shape` field. If the `geo_point` field contains an array, `geohash_grid`
aggregates all the array values.
```

Comment (talevy) [reply docs/reference/aggregations/bucket/geohashgrid-aggregation.asciidoc]: 😄

Comment (talevy) [reply docs/reference/aggregations/bucket/geohashgrid-aggregation.asciidoc]: was too haste over here. good catch

**Pull request merged**